<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!-- Essential parameters -->
  <arg name="vehicle_model" default="sample_vehicle" description="vehicle model name"/>
  <arg name="sensor_model" default="sample_sensor_kit" description="sensor model name"/>
  <arg name="rviz_config" default="$(find-pkg-share autoware_test_utils)/rviz/psim_test_map.rviz" description="rviz config"/>
  <arg name="use_sim_time" default="true"/>

  <include file="$(find-pkg-share autoware_launch)/launch/planning_simulator.launch.xml">
    <arg name="map_path" value="$(find-pkg-share autoware_test_utils)/test_map/intersection"/>
    <arg name="vehicle_model" value="$(var vehicle_model)"/>
    <arg name="sensor_model" value="$(var sensor_model)"/>
    <arg name="use_sim_time" value="$(var use_sim_time)"/>
    <arg name="rviz_config" value="$(var rviz_config)"/>
  </include>

  <node pkg="autoware_test_utils" exec="topic_snapshot_saver" name="topic_snapshot_saver">
    <param name="map_path" value="package://autoware_test_utils/test_map/intersection/lanelet2_map.osm"/>
    <param name="config_path" value="package://autoware_test_utils/config/sample_topic_snapshot.yaml"/>
  </node>
</launch>
